### PR TITLE
🔨 Refactor `Engine.predict` method

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## 📝 Description

- Addresses https://github.com/openvinotoolkit/anomalib/issues/1768

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](../docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/cli/cli.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 from jsonargparse import ActionConfigFile, ArgumentParser, Namespace
 from jsonargparse._actions import _ActionSubCommands
+from lightning.pytorch.core.datamodule import LightningDataModule
```
Comment (samet-akcay) [inline src/anomalib/cli/cli.py:16]: After https://github.com/openvinotoolkit/anomalib/pull/1748, we may not have lightning installed initially. Does this consider this?

Comment (ashwinvaidya17) [reply src/anomalib/cli/cli.py]: Ah good point! I have changed the import location

File: src/anomalib/cli/cli.py [hunk 2]
```diff
@@ -296,6 +297,9 @@ def instantiate_classes(self) -> None:
             # the minor change here is that engine is instantiated instead of trainer
             self.config_init = self.parser.instantiate_classes(self.config)
             self.datamodule = self._get(self.config_init, "data")
+            if isinstance(self.datamodule, Dataset):
+                param = {f"{self.config.subcommand}_dataset": self.datamodule}
```
Comment (samet-akcay) [inline src/anomalib/cli/cli.py:301]: a bit super minor and opinionated, but was wondering whether `kwargs` would be a better naming here or not?

**Review by samet-akcay — APPROVED** [review]
thanks!

**Review by djdameln — COMMENTED** [review]
Are you going to make the same changes for the other entrypoints? In the following for example,

```python
    def fit(
        self,
        model: AnomalyModule,
        train_dataloaders: TRAIN_DATALOADERS | AnomalibDataModule | None = None,
        val_dataloaders: EVAL_DATALOADERS | None = None,
        datamodule: AnomalibDataModule | None = None,
        ckpt_path: str | None = None,
    ) -> None:
```
it would be more consistent if `train_dataloaders` is only allowed to be a (list of) dataloader(s)

**Comment by ashwinvaidya17** [comment]
> Are you going to make the same changes for the other entrypoints? In the following for example,
> 
> ```python
>     def fit(
>         self,
>         model: AnomalyModule,
>         train_dataloaders: TRAIN_DATALOADERS | AnomalibDataModule | None = None,
>         val_dataloaders: EVAL_DATALOADERS | None = None,
>         datamodule: AnomalibDataModule | None = None,
>         ckpt_path: str | None = None,
>     ) -> None:
> ```
> 
> it would be more consistent if `train_dataloaders` is only allowed to be a (list of) dataloader(s)

Ah good point! I'll update those as well

**Pull request merged**